src/mesh/RTT_Draco_Mesh_Reader.cc

```diff
@@ -0,0 +1,85 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   mesh/RTT_Draco_Mesh_Reader.cc
+ * \author Ryan Wollaeger <wollaeger@lanl.gov>
+ * \date   Friday, Jul 13, 2018, 08:38 am
+ * \brief  RTT_Draco_Mesh_Reader header file.
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include "RTT_Draco_Mesh_Reader.hh"
+
+namespace rtt_mesh {
+
+//---------------------------------------------------------------------------//
+// CONSTRUCTOR
+//---------------------------------------------------------------------------//
+/*!
+ * \brief RTT_Draco_Mesh_Reader constructor.
+ *
+ * \param[in] filename_ name of file to be parsed
+ */
+RTT_Draco_Mesh_Reader::RTT_Draco_Mesh_Reader(std::string filename_)
+    : filename(filename_) {
+  // check for valid file name
+  Insist(filename_.size() > 0, "No file name supplied.");
+  // \todo: add more filename criteria?
```

Comment:
You can try something like:

```
std::ofstream inputfile(filename_.c_str());
Insist(inputfile().is_open(), "Failed to find or open specified mesh file.");
inputfile.close();
````

or delay this responsibility until the mesh file is actually read.

Reply:
Thank you for the recommendation - this has been added to the `read_mesh()` function, as in the X3D version.